op-e2e/config/init.go

```diff
@@ -0,0 +1,135 @@
+package config
+
+import (
+	"flag"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/state"
+)
+
+var (
+	// L1Allocs represents the L1 genesis block state.
+	L1Allocs *state.Dump
+	// L1Deployments maps contract names to accounts in the L1
+	// genesis block state.
+	L1Deployments *genesis.L1Deployments
+	// DeployConfig represents the deploy config used by the system.
+	DeployConfig *genesis.DeployConfig
+)
+
+// Init testing to enable test flags
+var _ = func() bool {
+	testing.Init()
+	return true
+}()
+
+func init() {
+	var l1AllocsPath, l1DeploymentsPath, deployConfigPath string
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+	root, err := findMonorepoRoot(cwd)
+	if err != nil {
+		panic(err)
+	}
+
+	defaultL1AllocsPath := filepath.Join(root, ".devnet", "allocs-l1.json")
+	defaultL1DeploymentsPath := filepath.Join(root, ".devnet", "addresses.json")
+	defaultDeployConfigPath := filepath.Join(root, "packages", "contracts-bedrock", "deploy-config", "devnetL1.json")
+
+	flag.StringVar(&l1AllocsPath, "l1-allocs", defaultL1AllocsPath, "")
+	flag.StringVar(&l1DeploymentsPath, "l1-deployments", defaultL1DeploymentsPath, "")
+	flag.StringVar(&deployConfigPath, "deploy-config", defaultDeployConfigPath, "")
+	flag.Parse()
+
+	if err := allExist(l1AllocsPath, l1DeploymentsPath, deployConfigPath); err != nil {
+		return
+	}
+
+	L1Allocs, err = genesis.NewStateDump(l1AllocsPath)
+	if err != nil {
+		panic(err)
+	}
+	L1Deployments, err = genesis.NewL1Deployments(l1DeploymentsPath)
+	if err != nil {
+		panic(err)
+	}
+	DeployConfig, err = genesis.NewDeployConfig(deployConfigPath)
+	if err != nil {
+		panic(err)
+	}
+
+	// Do not use clique in the in memory tests. Otherwise block building
+	// would be much more complex.
+	DeployConfig.L1UseClique = false
+	// Set the L1 genesis block timestamp to now
+	DeployConfig.L1GenesisBlockTimestamp = hexutil.Uint64(time.Now().Unix())
+	DeployConfig.FundDevAccounts = true
+
+	// Get the storage layout
+	account := L1Allocs.Accounts[L1Deployments.OptimismPortalProxy]
+	if len(account.Code) == 0 {
+		panic("portal proxy doesn't exist")
+	}
+
+	layout, err := bindings.GetStorageLayout("OptimismPortal")
+	if err != nil {
+		panic(err)
+	}
+	entry, err := layout.GetStorageLayoutEntry("params")
+	if err != nil {
+		panic(err)
+	}
+	slot := common.BigToHash(big.NewInt(int64(entry.Slot)))
```

Comment:
not ideal that we have to do this, it also means we will have to do it in the python code. ideally we could do it if we were using anvil but it doesn't support the rpc we need. it has an alt state dumping rpc that i'm not exactly sure how to work with to be honest, its binary + compressed, would need to look into it more

Reply:
fixed this issue